src/Model/PhasorDynamics/SystemModel.hpp

```diff
@@ -487,39 +507,84 @@ namespace GridKit
         }
       }
 
+      /**
+       * @brief Add bus
+       *
+       */
       void addBus(bus_type* bus)
       {
+        IdxT gridkit_bus_id                  = static_cast<IdxT>(buses_.size());
+        gridkit_bus_indices_[gridkit_bus_id] = bus->busID();
         buses_.push_back(bus);
       }
 
+      /**
+       * @brief Add signal
+       *
+       */
       void addSignal(signal_type* signal)
       {
+        IdxT gridkit_signal_id                     = static_cast<IdxT>(signals_.size());
+        gridkit_signal_indices_[gridkit_signal_id] = signal->signalId();
         signals_.push_back(signal);
       }
 
+      /**
+       * @brief Add component
+       *
+       */
       void addComponent(component_type* component)
       {
+        IdxT gridkit_component_id = static_cast<IdxT>(components_.size());
+        // No user-facing component_id for now, but we could add a map to the disambiguation_string
+        component->setGridKitComponentID(gridkit_component_id);
```

Comment:
Components are identified by the "disambiguation string", see [here](https://github.com/ORNL/GridKit/blob/develop/src/Model/PhasorDynamics/ComponentData.hpp#L45). Come to think of it, the user facing bus ID is integer but is typically used as a string.

@abirchfield, do we need to add component (device) IDs to the JSON input file format?